src/screens/OtherProfile/index.js

```diff
@@ -232,33 +234,16 @@ const OtherProfile = () => {
     }
   };
 
-  const gotoChatRoom = async () => {
-    const type = 'messaging';
-    const sort = [{last_message_at: -1}];
-
-    const members = [profile.myProfile.user_id, dataMain.user_id];
-    const filter = {type, members: {$eq: members}};
-
-    const clientChat = await client.client;
-
-    const findChannels = await clientChat.queryChannels(filter, sort, {
-      watch: true,
-      state: true
-    });
-
-    setChannel(findChannels[0], dispatchChannel);
-    await navigation.navigate('ChatDetailPage');
-  };
-
   const sendSignedDM = async () => {
     try {
       setLoadingSendDM(true);
       const signedMParams = {
         user_id: dataMain.user_id,
         message: dmChat
       };
-      await sendSignedDMOtherProfile(signedMParams);
-      await gotoChatRoom();
+      const response = await sendSignedDMOtherProfile(signedMParams);
+      const newResponse = {...response, members: response?.message?.members};
+      await saveChatFromOtherProfile(newResponse, 'sent', true, SIGNED);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The old code was navigating to the chat room after sending a signed DM. The new code is saving the chat from the other profile but not navigating to the chat room. If the navigation to the chat room is still required, you should add it back.
```diff
- await sendSignedDMOtherProfile(signedMParams);
- await gotoChatRoom();
+ const response = await sendSignedDMOtherProfile(signedMParams);
+ const newResponse = {...response, members: response?.message?.members};
+ await saveChatFromOtherProfile(newResponse, 'sent', true, SIGNED);
+ // Add navigation to chat room if needed
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->